src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -156,6 +222,673 @@ MultiRouterPlanCreate(Query *originalQuery, Query *query,
 
 
 /*
+ * Creates a router plan for INSERT ... SELECT queries which could consists of
+ * multiple tasks.
+ *
+ * The function never returns NULL, it errors out if cannot create the multi plan.
+ */
+static MultiPlan *
+CreateMultiTaskRouterPlan(Query *originalQuery,
```

Comment:
Afaics this is really INSERT ... SELECT specific, and not a more general "multi task router plan" function.